open-cas-linux-21.06.0.0541 build failed

ENV：
Operating System: Ubuntu 16.04.5.3 LTS
Kernel: Linux 4.15.0-58-generic x86_64
Architecture: x86-64
CAS VERSION：open-cas-linux-21.06.0.0541


compile failed
./configure -- ok
make  --error

```
cd modules && make
make[1]: Entering directory '/root/open-cas-linux-21.06.0.0541.release/modules'
make[2]: Entering directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[3]: Entering directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
rm -r  /root/open-cas-linux-21.06.0.0541.release/modules/.tmp_versions
make[3]: Leaving directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[2]: Leaving directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[2]: Entering directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[3]: Entering directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[3]: Nothing to be done for 'distcleandir'.
make[3]: Leaving directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
make[2]: Leaving directory '/root/open-cas-linux-21.06.0.0541.release/ocf'
cd "/lib/modules/"4.15.0-58-generic"/build" && make M=/root/open-cas-linux-21.06.0.0541.release/modules modules
make[2]: Entering directory '/usr/src/linux-headers-4.15.0-58-generic'
  CC [M]  /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.o
In file included from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/linux_kernel_version.h:44:0,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/ocf_env.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_def_priv.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.c:7:
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h: In function ‘cas_bdev_exist’:
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h:9:36: error: passing argument 2 of ‘lookup_bdev’ makes integer from pointer without a cast [-Werror=int-conversion]
         result = lookup_bdev(path, &dev);
                                    ^
In file included from ./include/linux/cgroup.h:17:0,
                 from ./include/linux/kthread.h:7,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/linux_kernel_version.h:18,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/ocf_env.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_def_priv.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.c:7:
./include/linux/fs.h:2575:29: note: expected ‘int’ but argument is of type ‘dev_t * {aka unsigned int *}’
 extern struct block_device *lookup_bdev(const char *, int mask);
                             ^
In file included from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/linux_kernel_version.h:44:0,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/ocf_env.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_def_priv.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.c:7:
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h:9:16: error: assignment makes integer from pointer without a cast [-Werror=int-conversion]
         result = lookup_bdev(path, &dev);
                ^
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h: In function ‘cas_bdev_match’:
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h:18:36: error: passing argument 2 of ‘lookup_bdev’ makes integer from pointer without a cast [-Werror=int-conversion]
         result = lookup_bdev(path, &dev);
                                    ^
In file included from ./include/linux/cgroup.h:17:0,
                 from ./include/linux/kthread.h:7,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/linux_kernel_version.h:18,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/ocf_env.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_def_priv.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.c:7:
./include/linux/fs.h:2575:29: note: expected ‘int’ but argument is of type ‘dev_t * {aka unsigned int *}’
 extern struct block_device *lookup_bdev(const char *, int mask);
                             ^
In file included from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/linux_kernel_version.h:44:0,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/ocf_env.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_def_priv.h:10,
                 from /root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.c:7:
/root/open-cas-linux-21.06.0.0541.release/modules/generated_defines.h:18:16: error: assignment makes integer from pointer without a cast [-Werror=int-conversion]
         result = lookup_bdev(path, &dev);
                ^
cc1: all warnings being treated as errors
scripts/Makefile.build:330: recipe for target '/root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.o' failed
make[4]: *** [/root/open-cas-linux-21.06.0.0541.release/modules/cas_cache/src/ocf/ocf_io.o] Error 1
scripts/Makefile.build:604: recipe for target '/root/open-cas-linux-21.06.0.0541.release/modules/cas_cache' failed
make[3]: *** [/root/open-cas-linux-21.06.0.0541.release/modules/cas_cache] Error 2
Makefile:1562: recipe for target '_module_/root/open-cas-linux-21.06.0.0541.release/modules' failed
make[2]: *** [_module_/root/open-cas-linux-21.06.0.0541.release/modules] Error 2
make[2]: Leaving directory '/usr/src/linux-headers-4.15.0-58-generic'
Makefile:48: recipe for target 'default' failed
make[1]: *** [default] Error 2
make[1]: Leaving directory '/root/open-cas-linux-21.06.0.0541.release/modules'
Makefile:22: recipe for target 'modules' failed
make: *** [modules] Error 2
```
in same machine build open-cas-linux-21.03.0.0496.release is ok

--- a/configure.d/Makefile
+++ b/configure.d/Makefile
@@ -8,6 +8,14 @@ obj-m += test_mod.o
 
 MAKE_DIR:=$(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
 
+check_cflag=$(shell echo "" | \
+	gcc -c -xc ${1} -o /dev/null - 2>/dev/null; \
+	if [ $$? -eq 0 ]; then echo 1; else echo 0; fi; )
+
+ifeq ($(call check_cflag,-Werror=int-conversion), 1)
+EXTRA_CFLAGS += -Werror=int-conversion
+endif
+
 all:
 	make -C $(KERNEL_DIR) M=$(MAKE_DIR) modules
 clean:

